core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -409,6 +418,7 @@ private Function compileFilter(
         if (condition) {
             ObjList<Function> workerFilters = new ObjList<>();
             for (int i = 0; i < workerCount; i++) {
+                restoreWhereClause(filterExpr);//restore original filters in node query models
```

Comment:
nit: `restoreWhereClause` should use start index `1` for accessing join models, `model.joinModels.get(0) == model` all the time

Reply:
fixed in https://github.com/questdb/questdb/pull/2958/commits/fdd798b752780c214589698d1cab4711e67af2f1